tripal/tripal
# Ensure TripalFields can be added through UI and add some tests

# Bug Fix

### Issue #1966 


### Tripal Version: 4

## Description


In the linked issue @dsenalik pointed out that TripalFields could not be added throguh the UI. I was already doing some automated tests for TripalFields in another branch so added the fix to that issue in with my testing to create this PR.

## Testing?




1. Tripal -> Page Structure -> Project -> Manage Fields
2. +Create a new field -> Tripal Fields -> Continue
3. Label enter anything
4. Check any of the four field types -> Continue
5. Confirm the new field is now listed in the fields for project.
6. Create a project with a number of values including a value for your new Tripal field instance and confirm it saves.
7. Edit this project, change a bunch of values including the value of your new TripalField and confirm it is updated.
8. Also create and edit content where the TripalField does not have a value entered.

### Automated Testing

This PR adds some basic testing for TripalFields. The tests in Drupal\Tests\tripal\Kernel\TripalField\TripalFieldTypeSettingsTest check that the subform for the field storage settings can be built. This test fails without the fix.

This PR also adds some basic testing of the Tripal field types in Drupal\Tests\tripal\Kernel\TripalField\TripalFieldTypeCRUDTest although it currently only tests create. Tests for edit and delete will be coming in another PR.

<img width="1788" alt="Screenshot 2024-09-03 at 2 17 06 PM" src="https://github.com/user-attachments/assets/715d2065-fed9-43ea-bc97-bb548b0d8f83">
<img width="1514" alt="Screenshot 2024-09-09 at 10 34 57 AM" src="https://github.com/user-attachments/assets/e6f761ff-65ba-4df0-8ca6-1f01801607a2">
